### ISSUE TITLE
Re-release with Helm 3.12.2 (3.12.1 cannot download charts)

### ISSUE THREAD

**Issue by asmacdo** [hashicorp/terraform-provider-helm#1302]
I am a terraform noob, sorry if this isn't in the right place! Originally: https://github.com/awslabs/data-on-eks/issues/376

While following [this blog](https://aws.amazon.com/blogs/containers/building-multi-tenant-jupyterhub-platforms-on-amazon-eks/) to deploy the jupyterhub blueprint , Terraform fails with the error:

```
module.eks_blueprints_addons.module.karpenter.helm_release.this[0]: Creating...
╷
│ Error: could not download chart: unexpected status from HEAD request to https://public.ecr.aws/v2/karpenter/karpenter/manifests/v0.32.1: 400 Bad Request
│ 
│   with module.eks_blueprints_addons.module.karpenter.helm_release.this[0],
│   on .terraform/modules/eks_blueprints_addons.karpenter/main.tf line 9, in resource "helm_release" "this":
│    9: resource "helm_release" "this" {
│ 
╵
```

I was eventually able to replicate this with Helm `3.13.1`, but 3.13.2 works as expected:

```
helm install oci://public.ecr.aws/karpenter/karpenter  --version v0.32.1 --generate-name

Error: INSTALLATION FAILED: unexpected status from HEAD request to https://public.ecr.aws/v2/karpenter/karpenter/manifests/v0.32.1: 400 Bad Request
```

However, the local version of helm didn't seem to affect Terraform, so I'm guessing this provider needs to be re-released with https://github.com/hashicorp/terraform-provider-helm/pull/1300/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R13

### Terraform, Provider, Kubernetes and Helm Versions
terraform -v
Terraform v1.6.5
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v5.27.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.2
+ provider registry.terraform.io/hashicorp/helm v2.12.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.24.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.9.2
+ provider registry.terraform.io/hashicorp/tls v4.0.5
```
Terraform version:
Provider version:
Kubernetes version:
```

### Steps to Reproduce
`git clone https://github.com/awslabs/data-on-eks.git
cd data-on-eks/ai-ml/jupyterhub && chmod +x install.sh`

### Expected Behavior

Helm installs chart

### Actual Behavior

Error: INSTALLATION FAILED: unexpected status from HEAD request to https://public.ecr.aws/v2/karpenter/karpenter/manifests/v0.32.1: 400 Bad Request

### References

Looks like this version bump was just fixed, all we need is a new release? 
https://github.com/hashicorp/terraform-provider-helm/pull/1300/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R13

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by asmacdo** [comment]
Thanks, should have hit refresh ;)

Fixed by: https://github.com/hashicorp/terraform-provider-helm/pull/1301

**Comment by arybolovlev** [comment]
Hi @asmacdo,

Thanks for raising this. I will close this issue since it is a duplicate of https://github.com/hashicorp/terraform-provider-helm/issues/1299 and since it was closed as resolved by merging PR you might have missed it.

I have merged all changes and `v2.12.1` will be available shortly.

Thank you for the detailed issue description though. 😊
